src/main/java/com/mycompany/mszczepienia/service/AuthService.java

```diff
@@ -105,4 +102,12 @@ public RefreshJwtResponseDto refreshToken(RefreshJwtRequestDto refreshJwtRequest
                 .refreshToken(refreshToken)
                 .build();
     }
+    @Transactional
+    public UpdatePasswordDto updatePassword(UpdatePasswordDto forgotPasswordDto){
+        User user = userRepository.findByEmail(forgotPasswordDto.getEmail()).orElseThrow(() ->
+                    new UserNotFoundException("Reset password", "There is no user with this email"));
+        user.setPassword(passwordEncoder.encode(forgotPasswordDto.getPassword()));
+        userRepository.save(user);
+        return modelMapper.map(user, UpdatePasswordDto.class);
```

Comment:
There's no need to return anything from this method - we're only changing the password and it should never be exposed anywhere.